pkg/sql/plan2/function/builtin/multi/floor.go

```diff
@@ -0,0 +1,161 @@
+	"github.com/matrixorigin/matrixone/pkg/vectorize/floor"
+	"github.com/matrixorigin/matrixone/pkg/vm/process"
+)
+
+// floor function's evaluation for arguments: [uint64]
+func FloorUInt64(vecs []*vector.Vector, proc *process.Process) (*vector.Vector, error) {
+	digits := int64(0)
+	vs := vecs[0].Col.([]uint64)
+	vec, err := process.Get2(proc, 8*int64(len(vs)), types.Type{Oid: types.T_uint64, Size: 8})
+	if err != nil {
+		return nil, err
+	}
+	rs := encoding.DecodeUint64Slice(vec.Data)
+	rs = rs[:len(vs)]
+	vec.Col = rs
+	nulls.Set(vec.Nsp, vecs[0].Nsp)
+	vector.SetCol(vec, floor.FloorUint64(vs, rs, digits))
+	if vecs[0].IsConst {
+		vec.IsConst = true
+		vec.Length = vecs[0].Length
+	}
+	return vec, nil
+}
+
+// floor function's evaluation for arguments: [int64]
+func FloorInt64(vecs []*vector.Vector, proc *process.Process) (*vector.Vector, error) {
+	digits := int64(0)
+	vs := vecs[0].Col.([]int64)
+	vec, err := process.Get(proc, 8*int64(len(vs)), types.Type{Oid: types.T_int64, Size: 8})
+	if err != nil {
+		return nil, err
+	}
+	rs := encoding.DecodeInt64Slice(vec.Data)
+	rs = rs[:len(vs)]
+	vec.Col = rs
+	nulls.Set(vec.Nsp, vecs[0].Nsp)
+	vector.SetCol(vec, floor.FloorInt64(vs, rs, digits))
+	if vecs[0].IsConst {
+		vec.IsConst = true
+		vec.Length = vecs[0].Length
+	}
+	return vec, nil
+}
+
+// floor function's evaluation for arguments: [uint64, int64]
+func FloorUInt64Int64(vecs []*vector.Vector, proc *process.Process) (*vector.Vector, error) {
+	digits := int64(0)
+	vs := vecs[0].Col.([]uint64)
+	// if the second paramter is null ,show error
+	if !vecs[1].IsConst || vecs[1].Typ.Oid != types.T_int64 || nulls.Any(vecs[1].Nsp) {
+		return nil, errors.New("the second argument of the round function must be an int64 constant")
+	}
+	digits = vecs[1].Col.([]int64)[0]
+	vec, err := process.Get2(proc, 8*int64(len(vs)), types.Type{Oid: types.T_uint64, Size: 8})
+	if err != nil {
+		return nil, err
+	}
+	rs := encoding.DecodeUint64Slice(vec.Data)
+	rs = rs[:len(vs)]
+	vec.Col = rs
+	nulls.Set(vec.Nsp, vecs[0].Nsp)
+	vector.SetCol(vec, floor.FloorUint64(vs, rs, digits))
+	if vecs[0].IsConst {
+		vec.IsConst = true
+		vec.Length = vecs[0].Length
+	}
+	return vec, nil
+}
+
+// floor function's evaluation for arguments: [float64]
+func FloorFloat64(vecs []*vector.Vector, proc *process.Process) (*vector.Vector, error) {
+	digits := int64(0)
+	vs := vecs[0].Col.([]float64)
+	vec, err := process.Get(proc, 8*int64(len(vs)), types.Type{Oid: types.T_float64, Size: 8})
+	if err != nil {
+		return nil, err
+	}
+	rs := encoding.DecodeFloat64Slice(vec.Data)
+	rs = rs[:len(vs)]
+	vec.Col = rs
+	nulls.Set(vec.Nsp, vecs[0].Nsp)
+	vector.SetCol(vec, floor.FloorFloat64(vs, rs, digits))
+	if vecs[0].IsConst {
+		vec.IsConst = true
+		vec.Length = vecs[0].Length
+	}
+	return vec, nil
+}
+
+// floor function's evaluation for arguments: [int64, int64]
+func FloorInt64Int64(vecs []*vector.Vector, proc *process.Process) (*vector.Vector, error) {
+	digits := int64(0)
```

Comment:
I remember that the second param of function `Floor` should be a const value. (maybe I have misremembered.)
And if so, you should do some check here like
`if !vecs[1].IsConst {
  return error.
}`

Reply:
it seems that I have misremembered. you have done this work at function FloorFloat64Int64